### ISSUE TITLE
MongoDB fixture shouldn't attempt to delete system.users

### ISSUE THREAD

**Issue by jamesblackburn** [ClearcodeHQ/pytest-dbfixtures#118]
The mongodb_factory attempts to delete systems collections (apart from system.indexes).  It shouldn't delete any system.* collection:

```
____________________ ERROR at teardown of test_create_user _____________________

    def drop():
        for db in mongo_conn.database_names():
            for collection_name in mongo_conn[db].collection_names():
                if collection_name != 'system.indexes':
>                   mongo_conn[db][collection_name].drop()

File "/reggie-users/is/jblackburn/git/arctic/pytest_dbfixtures-0.11.0-py2.7.egg/pytest_dbfixtures/factories/mongo.py", line 139

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
File "/users/is/jblackburn/pyenvs/arctic/lib/python2.7/site-packages/pymongo-3.0.2_1_dce3f36-py2.7-linux-x86_64.egg/pymongo/collection.py", line 687
in drop
    self.__database.drop_collection(self.__name)
File "/users/is/jblackburn/pyenvs/arctic/lib/python2.7/site-packages/pymongo-3.0.2_1_dce3f36-py2.7-linux-x86_64.egg/pymongo/database.py", line 519
in drop_collection
    self.command("drop", _unicode(name), allowable_errors=["ns not found"])
File "/users/is/jblackburn/pyenvs/arctic/lib/python2.7/site-packages/pymongo-3.0.2_1_dce3f36-py2.7-linux-x86_64.egg/pymongo/database.py", line 454
in command
    codec_options, **kwargs)
File "/users/is/jblackburn/pyenvs/arctic/lib/python2.7/site-packages/pymongo-3.0.2_1_dce3f36-py2.7-linux-x86_64.egg/pymongo/database.py", line 366
in _command
    allowable_errors)
File "/users/is/jblackburn/pyenvs/arctic/lib/python2.7/site-packages/pymongo-3.0.2_1_dce3f36-py2.7-linux-x86_64.egg/pymongo/pool.py", line 184
in command
    codec_options, check, allowable_errors)
File "/users/is/jblackburn/pyenvs/arctic/lib/python2.7/site-packages/pymongo-3.0.2_1_dce3f36-py2.7-linux-x86_64.egg/pymongo/network.py", line 54
in command
    helpers._check_command_response(response_doc, msg, allowable_errors)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

response = {'code': 12502, 'errmsg': "exception: can't drop system ns", 'msg': 'indexes dropped for collection', 'nIndexesWas': 2.0, ...}
msg = "command SON([('drop', u'system.users')]) on namespace admin.$cmd failed: %s"
allowable_errors = ['ns not found']

    def _check_command_response(response, msg=None, allowable_errors=None):
        """Check the response to a command for errors.
        """
        if "ok" not in response:
            # Server didn't recognize our message as a command.
            raise OperationFailure(response.get("$err"),
                                   response.get("code"),
                                   response)
    
        # TODO: remove, this is moving to _check_gle_response
        if response.get("wtimeout", False):
            # MongoDB versions before 1.8.0 return the error message in an "errmsg"
            # field. If "errmsg" exists "err" will also exist set to None, so we
            # have to check for "errmsg" first.
            raise WTimeoutError(response.get("errmsg", response.get("err")),
                                response.get("code"),
                                response)
    
        if not response["ok"]:
    
            details = response
            # Mongos returns the error details in a 'raw' object
            # for some errors.
            if "raw" in response:
                for shard in itervalues(response["raw"]):
                    # Grab the first non-empty raw error from a shard.
                    if shard.get("errmsg") and not shard.get("ok"):
                        details = shard
                        break
    
            errmsg = details["errmsg"]
            if allowable_errors is None or errmsg not in allowable_errors:
    
                # Server is "not master" or "recovering"
                if (errmsg.startswith("not master")
                        or errmsg.startswith("node is recovering")):
                    raise NotMasterError(errmsg)
    
                # Server assertion failures
                if errmsg == "db assertion failure":
                    errmsg = ("db assertion failure, assertion: '%s'" %
                              details.get("assertion", ""))
                    raise OperationFailure(errmsg,
                                           details.get("assertionCode"),
                                           response)
    
                # Other errors
                code = details.get("code")
                # findAndModify with upsert can raise duplicate key error
                if code in (11000, 11001, 12582):
                    raise DuplicateKeyError(errmsg, code, response)
                elif code == 50:
                    raise ExecutionTimeout(errmsg, code, response)
    
                msg = msg or "%s"
>               raise OperationFailure(msg % errmsg, code, response)
E               OperationFailure: command SON([('drop', u'system.users')]) on namespace admin.$cmd failed: exception: can't drop system ns
```

**Comment by fizyk** [comment]
What version of MongoDB do you use? Is it some new one, or old? We haven't used Mongo for a while and hadn't had the problem when we did use.

**Comment by jamesblackburn** [comment]
I'm using 2.4.6. I have a test that creates users - which I guess creates the users collection. 

Would also be a useful feature to allow creating function level mongodb fixtures.

**Comment by pwilczynskiclearcode** [comment]
Solved by version 0.12.0.
